### ISSUE TITLE
Enums of boolean types should be reflected as a zod literal

### ISSUE THREAD

**Issue by saphewilliam** [kubb-labs/kubb#1222]
### What version of `kubb` is running?

3.0.0-alpha.20

### What platform is your computer?

Linux

### What version of external packages are you using(`@tanstack-query`, `MSW`, `React`, `Vue`, ...)

typescript 5.6.2, zod 3.23.8

### What steps can reproduce the bug?

Given this (somewhat contrived, but based on a real example) kubb.config.ts file:
```ts
import { defineConfig } from '@kubb/core';
import { pluginOas } from '@kubb/plugin-oas';
import { pluginTs } from '@kubb/plugin-ts';
import { pluginZod } from '@kubb/plugin-zod';

export default defineConfig({
  name: 'Test',
  root: '.',
  input: {
    data: JSON.stringify({
      openapi: '3.0.0',
      components: {
        schemas: {
          Binary: {
            type: 'object',
            properties: {
              value: {
                type: 'boolean',
                enum: [true, false],
              },
            },
            required: ['value'],
          },
        },
      },
    }),
  },
  output: {
    path: './output',
  },
  plugins: [
    pluginOas({
      output: { path: 'oas' },
      validate: true,
    }),
    pluginTs({
      output: { path: 'models' },
    }),
    pluginZod({
      output: { path: 'schemas' },
      typed: true,
    }),
  ],
});
```

Run the generator and observe TypeScript error in the generated code

```ts
import type { Binary } from "../models/Binary.ts";
import { z } from "zod";

export const binarySchema = z.object({ 
  "value": z.enum(["true", "false"]),
}) as z.ZodType<Binary>;
```

```txt
Conversion of type 'ZodObject<{ value: ZodEnum<["true", "false"]>; }, "strip", ZodTypeAny, { value: "true" | "false"; }, { value: "true" | "false"; }>' to type 'ZodType<Binary, ZodTypeDef, Binary>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  The types of '_type.value' are incompatible between these types.
    Type 'string' is not comparable to type 'BinaryValue'.
      Type 'string' is not comparable to type 'BinaryValue'.ts(2352)
```

### How often does this bug happen?

Every time

### What is the expected behavior?

The expected output is this:

```ts
import type { Binary } from "../models/Binary.ts";
import { z } from "zod";

export const binarySchema = z.object({ 
  "value": z.union([z.literal(true), z.literal(false)]),
  // or, for enums with 1 value: `"value": z.literal(true)`
}) as z.ZodType<Binary>;
```

This gets rid of the typescript error

### Swagger/OpenAPI file?

Included in the repro

### Additional information

Thanks for creating this amazing library and your continued work on improving it!

**Comment by stijnvanhulle** [comment]
Thanks for using Kubb and pointing this out. Appreciate testing and helping us out.

This will be fixed with Kubb `v2.26.5` and `v3.0.0-alpha.22`.

**Comment by saphewilliam** [comment]
Amazing! Thanks so much :)
